### DISCUSSION TITLE
How to dynamically generate and apply color classes?

### DISCUSSION THREAD

**Discussion by edgarasben** [tailwindlabs/tailwindcss#9961] [Help] [answered]
I am using Radix color scales where the “logic” for light/dark theming is directly embeded to color naming.

How can I apply multiple classes to one class dynamically without manyally doing it with @apply?

This means when I use a class:
`bg-primary-9`, I want classes `bg-light-primary-9 dark:bg-dark-primary-9` to be applied (mapping to the color name and the number).

Similarly, I want `bg-primary-alpha-9` to also apply as `bg-light-primary-alpha-9 dark:bg-dark-primary-alpha-9`.

**Comment by wongjn** [comment, marked as answer]
You could look at using CSS variables (declared in CSS or via plugins), assuming you're using `darkMode: 'class'`:
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --color-1: 255 255 255;
  /* … */
}
.dark :root {
  --color-1: 0 0 0;
  /* … */
}
```
Then in your Tailwind configuration, you could do:
```js
module.exports = {
  // …
  darkMode: 'class',
  theme: {
    extend: {
      colors: {
        'color-1': 'rgb(var(--color-1) / <alpha-value>)',
        // …
      },
    },
  },
};
```
[See more about this CSS variable syntax for colors in the documentation](https://tailwindcss.com/docs/customizing-colors#using-css-variables).

**Reply by edgarasben** [reply]
I've tried this approach with CSS variables. However, this will only give one color. As you can see in my example, I need to be able to "call" two colors with only using one color class. Not sure if I explained it correctly in my initial post.

Basically I want to write `bg-primary-9` but get a result if I wrote `bg-light-primary-9 dark:bg-dark-primary-9`. But I still need to be able to use `bg-light-primary-9` and `dark:bg-dark-primary-9` seperately when I need to.

**Reply by wongjn** [reply]
For the `.dark :root` rule, you can have the value be the `dark-primary-9` color and vice-versa.

**Reply by edgarasben** [reply]
But how can this work with `darkMode: 'media'`? Also I would prefer setting dark theme without CSS variables, because I import my color variables via JSON directly to the config file.

**Reply by wongjn** [reply]
Adjust the rule a little:
```css
@media (prefers-color-scheme: dark) {
  :root {
    --color-1: 0 0 0;
    /* … */
  }
}
```

**Reply by edgarasben** [reply]
@wongjn ok I think it works! Thank you!

css:
```css
@layer base {
  :root {
    --color-primary-1: 240, 33.3%, 99.4%, 1;
    --color-primary-2: 223, 100%, 98.6%, 1;
    --color-primary-3: 224, 100%, 97.1%, 1;
    --color-primary-4: 222, 92.3%, 94.9%, 1;
    --color-primary-5: 225, 85.4%, 92%, 1;
    --color-primary-6: 224, 81%, 87.6%, 1;
    --color-primary-7: 225, 78%, 82.2%, 1;
    --color-primary-8: 226, 75.4%, 74.5%, 1;
    --color-primary-9: 226, 70%, 55.5%, 1;
    --color-primary-10: 226, 58.9%, 51.4%, 1;
    --color-primary-11: 226, 54.8%, 45.1%, 1;
    --color-primary-12: 226, 62.8%, 16.9%, 1;
...
```

tailwind.config.js:
```js
  theme: {
    colors: {
      transparent: 'transparent',
      primary: {
        1: 'hsla(var(--color-primary-1))',
        2: 'hsla(var(--color-primary-2))',
        3: 'hsla(var(--color-primary-3))',
        4: 'hsla(var(--color-primary-4))',
        5: 'hsla(var(--color-primary-5))',
        6: 'hsla(var(--color-primary-6))',
        7: 'hsla(var(--color-primary-7))',
        8: 'hsla(var(--color-primary-8))',
        9: 'hsla(var(--color-primary-9))',
        10: 'hsla(var(--color-primary-10))',
        11: 'hsla(var(--color-primary-11))',
        12: 'hsla(var(--color-primary-12))',
...
```

**Reply by edgarasben** [reply]
I with though there was a way to automate this :D A bit of manual labour to do it in css and config.js.

**Comment by edgarasben** [comment]
I was also offered an alternative solution in Discord channel. This solution requires [tailwindcss-theme-swapper](https://github.com/crswll/tailwindcss-theme-swapper) Tailwind plugin, but it can be helpful if colors come from other js/json files, rather than using CSS variables as in the solution above.
